### PR TITLE
python3Packages.devtools: init at 0.6.1

### PR DESCRIPTION
**Pull request by jdahm**
###### Motivation for this change

Add "Python's missing debug print command and other development tools".

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129898` at af7edae9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.devtools</li>
    <li>python39Packages.devtools</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/devtools/default.nix:13:0:
```
   |
13 |     description = "Python's missing debug print command and other development tools";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/devtools/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.6.1";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1ms2kp1h395880k1bcn71ap53s33ffb2jxnzx3x2hpfkjisk0m50";
+  };
+
+  propagatedBuildInputs = [ pygments ];
+
+  # tests not distributed on pypi
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/devtools/default.nix:13]: There are GitHub releases present. Please use those and run the [tests](https://github.com/samuelcolvin/python-devtools/tree/master/tests).

File: pkgs/development/python-modules/devtools/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  pythonImportsCheck = [ "devtools" ];
+
+  meta = with lib; {
+    description = "Python's missing debug print command and other development tools";
+    homepage = "https://github.com/samuelcolvin/python-devtools";
```
Comment (fabaff) [inline pkgs/development/python-modules/devtools/default.nix:18]:
```suggestion
    homepage = "https://python-devtools.helpmanual.io/";
```

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -4787,6 +4787,12 @@
     githubId = 1982341;
     name = "Jack Cummings";
   };
+  jdahm = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:4790]: This should be part of a separate commit in this PR.

Comment (jdahm) [reply maintainers/maintainer-list.nix:4790]: Thanks for your very prompt and helpful feedback. I incorporated your comments above, merged all into a single PR, and made a second commit with just the maintainer addition. Was this what you intended? Thanks again.

File: pkgs/development/python-modules/devtools/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  propagatedBuildInputs = [ pygments ];
+
+  doCheck = true;
+  checkInputs = [ pytestCheckHook pytest-mock ];
+
+  meta = with lib; {
+    description =
+      "Python's missing debug print command and other development tools";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/devtools/default.nix:22]:
```suggestion
    description = "Python's missing debug print command and other development tools";
```

File: pkgs/development/python-modules/devtools/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "samuelcolvin";
+    repo = "python-${pname}";
+    rev = "v${version}";
+    sha256 = "0s1d2jwijini7y1a3318yhb98mh1mw4pzlfx2zck3a8nqw984ki3";
+  };
+
+  propagatedBuildInputs = [ pygments ];
+
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/devtools/default.nix:17]:
```suggestion
```
that is the default.

File: pkgs/development/python-modules/devtools/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ buildPythonPackage, pythonOlder, fetchFromGitHub, lib, pygments
+, pytestCheckHook, pytest-mock }:
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/devtools/default.nix:3]:
```suggestion
, pytestCheckHook, pytest-mock }:

buildPythonPackage rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please squash all the package related commits.

File: pkgs/development/python-modules/devtools/default.nix (hunk 4 above)
Comment (jdahm) [reply pkgs/development/python-modules/devtools/default.nix:22]: Thanks, I will make that change. I used `nixfmt` to format these. Is there a specific width setting that I should use in the future?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/devtools/default.nix]: > Is there a specific width setting that I should use in the future?

We don't really have one. Just keep it reasonable.
